lib/rubocop/cop/lint/duplicate_branch.rb

```diff
@@ -87,6 +103,27 @@ def offense_range(duplicate_branch)
             parent.source_range
           end
         end
+
+        def branches(node)
+          node.branches.compact
+        end
+
+        def consider_branch?(branch)
+          return true unless ignore_literal_branches?
+          return true unless branch.literal?
+          return false if branch.basic_literal?
+          return true if branch.xstr_type?
+
+          branch.each_descendant.any? do |node|
```

Comment:
I'm not 100% sure, but I would exclude hash literals however simple they are (same as arrays).
Are you allowing constants? I think we should.

Reply:
I added `IgnoreConstantBranches` to ignore constants now.

I think it's more consistent to let `IgnoreLiteralBranches` ignore all literals including hashes and arrays. @bbatsov what do you think?

Reply:
I agree. It's uncommon to have calculations done in literals, so I'd stick with a simple and consistent definition.

Reply:
I really don't think there should be a separate setting for constants.

Thinking about it more, in my mind it's actually not about being a literal but an already simple and well-defined object. As such a variable (not allowed currently, right?) / constant / integer / symbol are all acceptable. These involve no processing and no memory allocation. There's no good reason to factorize them, i.e. to store them / label them in one place before using them in multiple places. I'm not sure I have a name for this though. Atom?  Interned? Simple values?

Just a last example, before you decide to keep hashes and arrays:

```ruby
case
when method_a
  { hello: 'world',
    :does_it? => 'make sense',
    to: :accept_this,
  }
when method_b
  { hi: 'world',
    :does_it? => 'make sense',
    to: :accept_this,
  }
when method_c then
  { hello: 'world',
    :does_it? => 'make sense',
    to: :accept_this,
  }
end
```

Compare this to:

```ruby
case
when method_a then A
when method_b then B
when method_c then C
else B
end
```

Same with `A/B/C/B` being variables.

Reply:
Yeah, I totally get that those scenarios would be possible, but they are not very common and anything we define as "trivial" or whatever would carry some cognitive overhead as people have to understand our definition of trivial. I'd rather go with something that leaves no room for interpretation and solves the problem for or less. Plus, we're not changing the defaults. :-)

Reply:
Agree with @bbatsov. The initial idea here was just to make this less restrictive based on standard's issue, which I thought was valid. I've iterated on this three times now but I think we're getting into nitpicking so I'd rather scope hammer this back to the original goal.